utils/obsproc/NetCDFToIodaConverter2D.h

```diff
@@ -0,0 +1,318 @@
+#pragma once
+
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "eckit/config/LocalConfiguration.h"
+
+#include "ioda/Engines/HH.h"
+#include "ioda/Group.h"
+#include "ioda/ObsGroup.h"
+
+#include "oops/mpi/mpi.h"
+#include "oops/util/DateTime.h"
+#include "oops/util/Duration.h"
+#include "oops/util/Logger.h"
+#include "oops/util/missingValues.h"
+
+namespace gdasapp {
+  // A simple data structure to organize the info to provide to the ioda
+  // writter
+  struct IodaVars2d {
+    int location_;      // Number of observation per variable
+    int nVars_;         // number of obs variables, should be set to
+                       // for now in the children classes
+    int channel_;       // number of channels
+    int nfMetadata_;    // number of float metadata fields
+    int niMetadata_;    // number of int metadata fields
+
+    // Non optional metadata
+    Eigen::ArrayXf longitude_;  // geo-location_
+    Eigen::ArrayXf latitude_;   //     "
+    Eigen::Array<int64_t, Eigen::Dynamic, 1> datetime_;   // Epoch date in seconds
+    std::string referenceDate_;                           // Reference date for epoch time
+    Eigen::Array<int, Eigen::Dynamic, 1> channelNumber_;
+
+    // Obs info
+    Eigen::ArrayXf obsVal_;     // Observation value
+    Eigen::ArrayXf obsError_;   //      "      error
+    Eigen::ArrayXi preQc_;      // Quality control flag
+
+    // Optional metadata
+    Eigen::ArrayXXf floatMetadata_;                // Optional array of float metadata
+    std::vector<std::string> floatMetadataName_;  // String descriptor of the float metadata
+    Eigen::ArrayXXf intMetadata_;                  // Optional array of integer metadata
+    std::vector<std::string> intMetadataName_;    // String descriptor of the integer metadata
+
+    // Optional global attributes
+    std::map<std::string, std::string> strGlobalAttr_;
+
+    // Constructor
```

Comment:
you could just add the this constructor to the original iodavars struct, or have a case for channels in the original constructor.